dataverse-persistence/src/test/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetFieldValidatorTest.java

```diff
@@ -129,11 +129,12 @@ public void testIsValidAuthorIdentifierGnd() {
                 Arguments.of("", Option.none(), Option.none(), true),
                 Arguments.of("asdf", Option.none(), Option.none(), false),
                 Arguments.of("asdfgx", Option.none(), Option.none(), false),
-                Arguments.of("1999AD", Option.of(FieldType.DATE), Option.some(null), true),
-                Arguments.of("44BCE", Option.of(FieldType.DATE), Option.none(), true),
+                Arguments.of("1999AD", Option.of(FieldType.DATE), Option.some(null), false),
+                Arguments.of("1999", Option.of(FieldType.DATE), Option.none(), true),
+                Arguments.of("44BCE", Option.of(FieldType.DATE), Option.none(), false),
                 Arguments.of("2004-10-27", Option.of(FieldType.DATE), Option.none(), true),
                 Arguments.of("2002-08", Option.of(FieldType.DATE), Option.none(), true),
-                Arguments.of("[1999?]", Option.of(FieldType.DATE), Option.none(), true),
+                Arguments.of("[1999?]", Option.of(FieldType.DATE), Option.none(), false),
```

Comment:
I think that we should check more cases like:
what will happen for example if we have three digit year, for example 966
what if we enter 1999-1 or 1999-01-1
dates with minus sign -1999 -1999-01
What will happen if we enter year 10000
Check for supporting invalid month numbers like 1999-13
Check for supporting leap years 2020-02-29 vs 2019-02-29